app/scripts/lib/keyring-snaps-permissions.ts

```diff
@@ -0,0 +1,82 @@
+import {
+  SubjectType,
+  SubjectMetadataController,
+} from '@metamask/permission-controller';
+import { KeyringRpcMethod } from '@metamask/keyring-api';
+
+/**
+ * List of keyring methods MetaMask can call.
+ */
+const METAMASK_ALLOWED_METHODS: string[] = [
+  KeyringRpcMethod.ListAccounts,
+  KeyringRpcMethod.GetAccount,
+  KeyringRpcMethod.FilterAccountChains,
+  KeyringRpcMethod.DeleteAccount,
+  KeyringRpcMethod.ListRequests,
+  KeyringRpcMethod.GetRequest,
+  KeyringRpcMethod.SubmitRequest,
+  KeyringRpcMethod.RejectRequest,
+];
+
+/**
+ * List of keyring methods a dapp can call.
+ */
+const WEBSITE_ALLOWED_METHODS: string[] = [
+  KeyringRpcMethod.ListAccounts,
+  KeyringRpcMethod.GetAccount,
+  KeyringRpcMethod.CreateAccount,
+  KeyringRpcMethod.FilterAccountChains,
+  KeyringRpcMethod.UpdateAccount,
+  KeyringRpcMethod.DeleteAccount,
+  KeyringRpcMethod.ExportAccount,
+  KeyringRpcMethod.ListRequests,
+  KeyringRpcMethod.GetRequest,
+  KeyringRpcMethod.ApproveRequest,
+  KeyringRpcMethod.RejectRequest,
+];
+
+/**
+ * List of allowed protocols. On Flask, HTTP is also allowed for testing.
+ */
+const ALLOWED_PROTOCOLS: string[] = [
+  'https:',
+  ///: BEGIN:ONLY_INCLUDE_IN(build-flask)
+  'http:',
```

Comment:
Would be nice to test this, but I guess that's a bit difficult with the current implementation. You could move just this variable to a separate file and mock it in Jest, but that's not great either 😅.

Reply:
Yeah, I started to do something like this and came to the same conclusion 😄